lib/app/app.go

```diff
@@ -0,0 +1,184 @@
+package app
+
+import (
+	"flag"
+	"net"
+	"net/http"
+	_ "net/http/pprof"
+	"strconv"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/dedis/cothority/lib/logutils"
+	dbg "github.com/dedis/cothority/lib/debug_lvl"
+	"os"
+	"reflect"
+	"bytes"
+	"github.com/BurntSushi/toml"
+	"io/ioutil"
+	"path/filepath"
+)
+
+type FlagConfig struct {
+	Hostname    string // Hostname like server-0.cs-dissent ?
+	Logger      string // ip addr of the logger to connect to
+	PhysAddr    string // physical IP addr of the host
+	AmRoot      bool   // is the host root (i.e. special operations)
+	TestConnect bool   // Dylan-code to only test the connection and exit afterwards
+	Mode        string // ["server", "client"]
+	Name        string // Comes from deter.go:187 - "Name of the node"
+	Server      string // Timestamping servers to contact
+}
+
+// Initialize before 'init' so we can directly use the fields as parameters
+// to 'Flag'
+var Flags FlagConfig
```

Comment:
Since FlagConfig is holding all the flags we are passing to the executable, could we simply rename it "Flags" ?
There is too much "config" everywhere, we should restrict that as much as possible